### PR TITLE
Adding a recipe for cchardet

### PR DESCRIPTION
**Pull request by strets123**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1 +1,2 @@
 build_artefacts
+.idea/
```
Comment (jakirkham) [inline .gitignore]: Please break this out as a separate PR. Also please use `[skip ci]` in the commit message for that change. Thanks.

File: recipes/cchardet/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  fn: {{ name|lower }}-{{ version }}.tar.gz
+  url: https://files.pythonhosted.org/packages/14/4f/7570c170110a79290824a683f92043ecc048f851e57a2b2e223a9fe5e8c2/cchardet-{{ version }}.tar.gz
+  md5: {{ md5 }}
+
+build:
+  number: 2
```
Comment (jakirkham) [inline recipes/cchardet/meta.yaml]: Please set build number to `0`.

File: recipes/cchardet/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,53 @@
+{% set name = "cchardet" %}
+{% set version = "1.0.0" %}
+{% set md5 = "353f5ca2bc445cef5f103a9a7c2f7599" %}
```
Comment (jakirkham) [inline recipes/cchardet/meta.yaml]: Please use sha256 from [pypi.org]( https://pypi.org/ ).

File: recipes/cchardet/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  fn: {{ name|lower }}-{{ version }}.tar.gz
+  url: https://files.pythonhosted.org/packages/14/4f/7570c170110a79290824a683f92043ecc048f851e57a2b2e223a9fe5e8c2/cchardet-{{ version }}.tar.gz
```
Comment (jakirkham) [inline recipes/cchardet/meta.yaml]: Please use the URL from the [`example`]( https://github.com/conda-forge/staged-recipes/blob/4d2f7fe2712499ec394c8b5a118a8dd91e9ff843/recipes/example/meta.yaml#L18 ).

File: recipes/cchardet/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+test:
+  # Python imports
+  imports:
+    - cchardet
+
+
+
+about:
+  home: https://github.com/PyYoshi/cChardet
+  license: MIT
```
Comment (jakirkham) [inline recipes/cchardet/meta.yaml]: Please ask upstream to include a license file and add it to `MANIFEST.in`. This way we can package it and be in compliance with the MIT.

File: recipes/cchardet/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+source:
+  fn: {{ name|lower }}-{{ version }}.tar.gz
+  url: https://files.pythonhosted.org/packages/14/4f/7570c170110a79290824a683f92043ecc048f851e57a2b2e223a9fe5e8c2/cchardet-{{ version }}.tar.gz
+  md5: {{ md5 }}
+
+build:
+  number: 2
+  # If the installation is complex, or different between Unix and Windows, use separate bld.bat and build.sh files instead of this key.
+  # By default, the package will be built for Python 2.7, 3.4, and above, for all major OSs.
+  # Add the line "skip: True  # [py<34]" (for example) to limit to Python 3.4 and newer, or "skip: True  # [not win]" to limit to Windows.
```
Comment (jakirkham) [inline recipes/cchardet/meta.yaml]: Please drop these comments from the `example`.

File: recipes/cchardet/meta.yaml [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+
+
+about:
+  home: https://github.com/PyYoshi/cChardet
+  license: MIT
+  summary: 'Universal encoding detector. This library is faster than chardet.'
+
+# See
+# http://docs.continuum.io/conda/build.html for
+# more information about meta.yaml
```
Comment (jakirkham) [inline recipes/cchardet/meta.yaml]: Please drop these comments from the `example`.

File: recipes/cchardet/meta.yaml [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+
+extra:
+  recipe-maintainers:
+    # GitHub IDs for maintainers of the recipe.
+    # Always check with the people listed below if they are OK becoming maintainers of the recipe. (There will be spam!)
```
Comment (jakirkham) [inline recipes/cchardet/meta.yaml]: Please drop these comments from the `example`.

File: recipes/cchardet/meta.yaml [hunk 9]
```diff
@@ -0,0 +1,53 @@
...
+    - strets123
+    - zegami
```
Comment (jakirkham) [inline recipes/cchardet/meta.yaml]: @zegami, are you ok with being a maintainer here?

File: recipes/cchardet/meta.yaml [hunk 10]
```diff
@@ -0,0 +1,53 @@
...
+  run:
+    - python
+
+test:
+  # Python imports
+  imports:
+    - cchardet
+
+
+
```
Comment (jakirkham) [inline recipes/cchardet/meta.yaml]: Please drop to one blank line.

File: recipes/cchardet/meta.yaml [hunk 11]
```diff
@@ -0,0 +1,53 @@
...
+  build:
+    - python
+    - setuptools
+    - cython
+
+  run:
+    - python
+
+test:
+  # Python imports
```
Comment (jakirkham) [inline recipes/cchardet/meta.yaml]: Please drop these comments from the `example`.

File: recipes/cchardet/meta.yaml [hunk 12]
```diff
@@ -0,0 +1,53 @@
...
+  number: 2
+  # If the installation is complex, or different between Unix and Windows, use separate bld.bat and build.sh files instead of this key.
+  # By default, the package will be built for Python 2.7, 3.4, and above, for all major OSs.
+  # Add the line "skip: True  # [py<34]" (for example) to limit to Python 3.4 and newer, or "skip: True  # [not win]" to limit to Windows.
+  script: python setup.py install --single-version-externally-managed --record record.txt
+
+
+
+requirements:
+  build:
```
Comment (jakirkham) [inline recipes/cchardet/meta.yaml]: Please add `toolchain`. This configures our compilers. Details at that feedstock.

**Comment by strets123** [comment]
@jakirkham Thanks for the comments I will correct in the next few days, I have another 10 or 15 recipes to contribute, should each have a separate pull request? These recipes feed a closed source project so was not sure how to handle it. We have an empty package which holds all the requirements here: https://anaconda.org/zegami/zegami-dependencies but not sure if it is fine to push it all as one feedstock?

Is it OK to create recipes for forks off main repos or do you prefer official pypi versions of things?

**Comment by jakirkham** [comment]
Preferably they should all be separate PRs, but occasionally a couple related recipes may be included in one PR. Things to keep in mind: more recipes means a longer turnaround on CI, and PRs with more recipes may take longer to get reviewed. So there are some incentives for keeping PRs simple and easy. One recipe does that.

We strongly prefer official PyPI packages. We don't really want to get in the game of building multiple forks of a code base. This gets pretty messy in terms of maintenance and can create conflict. Would recommend PRing any changes you have to the upstream repos and working with their developers to get them incorporated.

**Comment by strets123** [comment]
OK, thanks for the advice

**Comment by jakirkham** [comment]
Appears this exists in a [feedstock]( https://github.com/conda-forge/cchardet-feedstock ) now. Going to close this out.

**Pull request closed without merging**